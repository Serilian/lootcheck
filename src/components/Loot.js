import React, {Component} from 'react';
import {connect} from 'react-redux';
import {fetchBitcoin} from "../store/actions/bitcoin";

export class Loot extends Component {

    componentDidMount() {
        this.props.fetchBitcoin();
    }

    computeBitcoin() {
        const {bitcoin} = this.props;
        if (!bitcoin.bpi) {
            return '';
        }

        return this.props.balance / parseInt(bitcoin.bpi.USD.rate.replace(',', ''), 10);

    }

    render() {
        return (
            <div>
                <h3 className={'bitcoin-balance'}>Bitcoin Balance: {this.computeBitcoin()}</h3>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    balance: state.balance,
    bitcoin: state.bitcoin
});

export default connect(mapStateToProps, {fetchBitcoin})(Loot);