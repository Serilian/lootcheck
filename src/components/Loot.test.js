import React from "react";
import {shallow, mount} from "enzyme";
import {Loot} from './Loot';

describe("Given wallet component", () => {
    const MOCK_BALANCE = 10;
    const MOCK_BITCOIN_DATA = {};


    let props = {balance: MOCK_BALANCE, bitcoin: MOCK_BITCOIN_DATA};

    let loot = shallow(<Loot {...props}/>);

    it("Then it renders correctly", () => {
        expect(loot.debug()).toMatchSnapshot();
    });

    describe("When mounted", () => {
        let mockFetchBitcoin = jest.fn();
        beforeEach(() => {
            props.fetchBitcoin = mockFetchBitcoin;
            loot = mount(<Loot {...props}/>);
        });

        it('dispatches `fetchBitcoin()` action', () => {
            expect(mockFetchBitcoin).toHaveBeenCalled();
        });

    });

    describe("when there are valid bitcoin props", () => {
        beforeEach(() => {
            props = {balance: 10, bitcoin: {bpi: {USD: {rate: `1,000`}}}};
            loot = shallow(<Loot {...props} />);
        });

        it('displays the value of bitcoin balance', () => {
            expect(loot.find(".bitcoin-balance").text()).toEqual(`Bitcoin Balance: 0.01`);
        });
    });


});