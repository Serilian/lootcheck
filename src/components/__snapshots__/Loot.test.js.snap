// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given wallet component Then it renders correctly 1`] = `
"<div>
  <h3 className=\\"bitcoin-balance\\">
    Bitcoin Balance: 
  </h3>
</div>"
`;
